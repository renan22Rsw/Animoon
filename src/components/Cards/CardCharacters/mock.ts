interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
