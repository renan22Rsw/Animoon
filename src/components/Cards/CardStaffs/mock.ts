interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
