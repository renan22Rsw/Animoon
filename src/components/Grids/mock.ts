interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
