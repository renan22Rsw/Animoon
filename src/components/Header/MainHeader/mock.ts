interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
