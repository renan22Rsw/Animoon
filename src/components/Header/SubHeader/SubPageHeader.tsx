import Image from "next/image";
import React from "react";
import TextArea from "../../TextArea/TextArea";

interface SubHeaderProps {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown?: string;
  yearsActive?: number[];
  description: string;
}

const SubPageHeader = ({
  name,
  nativeName,
  alternativeName,
  age,
  image,
  gender,
  month,
  day,
  bloodType,
  hometown,
  yearsActive,
  description,
}: SubHeaderProps) => {
  return (
    <div>
      <div className=" pb-[30px] pt-[20px] text-[#9FADB3]  ">
        <div className="max-w-[1200px]  hidden m-8 md:flex md:items-start lg:mx-10 2xl:mx-auto">
          <Image
            width={300}
            height={300}
            quality={100}
            src={image}
            alt="Anime-Image"
            style={{ height: "auto", width: "230px", objectFit: "cover" }}
            priority={true}
            className="rounded-md "
            data-testid="desktop-version"
          />
          <div className="my-4 mx-8">
            <h2 className="text-[#ADC0D2] text-3xl font-bold py-2">{name}</h2>
            <p
              className="md:text-sm lg:text-sm"
              data-testid="nativeName-desktop"
            >
              {nativeName && nativeName + ", "}
              {alternativeName ? alternativeName.join(", ") : ""}
            </p>

            <div className="my-8">
              {month || day ? (
                <h4 data-testid="desktop-birthday">
                  Birthday: {month + "/" + day}
                </h4>
              ) : (
                ""
              )}

              {age ? <h4 data-testid="desktop-age">Age: {age}</h4> : ""}

              {gender ? (
                <h4 data-testid="desktop-gender">Gender: {gender} </h4>
              ) : (
                ""
              )}

              {yearsActive
                ? yearsActive.map((yearActive, index) => (
                    <h4 key={index}>Years active: {yearActive}</h4>
                  ))
                : ""}
              {hometown ? (
                <h4 data-testid="desktop-hometown">Hometown: {hometown} </h4>
              ) : (
                ""
              )}
              {bloodType ? (
                <h4 data-testid="desktop-bloodType">Blood Type: {bloodType}</h4>
              ) : (
                ""
              )}
              <TextArea>{description}</TextArea>
            </div>
          </div>
        </div>

        <div className="text-center px-4 md:hidden  ">
          <h2 className="text-[#ADC0D2] text-2xl font-bold py-2">{name}</h2>
          <p data-testid="nativeName-mobile">
            {nativeName ? nativeName + ", " : ""}
            {alternativeName ? alternativeName.join(", ") : ""}
          </p>
          <div className="flex justify-center p-4  my-8">
            <Image
              width={130}
              height={130}
              src={image}
              alt="Anime-Image"
              style={{ height: "auto", width: "230px", objectFit: "cover" }}
              priority={true}
              className="rounded-md "
              data-testid="mobile-version"
            />
          </div>
          <div className="text-start mx-4">
            {month || day ? (
              <h4 data-testid="mobile-birthday">
                Birthday: {month + "/" + day}
              </h4>
            ) : (
              ""
            )}
            {age ? <h4 data-testid="mobile-age">Age: {age}</h4> : ""}
            {gender ? (
              <h4 data-testid="mobile-gender">Gender: {gender} </h4>
            ) : (
              ""
            )}
            {yearsActive
              ? yearsActive.map((yearActive, index) => (
                  <h4 key={index}>Years active: {yearActive}</h4>
                ))
              : ""}
            {hometown ? (
              <h4 data-testid="mobile-hometown">Hometown: {hometown} </h4>
            ) : (
              ""
            )}
            {bloodType ? (
              <h4 data-testid="mobile-bloodType">Blood Type: {bloodType}</h4>
            ) : (
              ""
            )}

            <TextArea>{description}</TextArea>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SubPageHeader;
