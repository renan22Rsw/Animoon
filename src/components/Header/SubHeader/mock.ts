interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
