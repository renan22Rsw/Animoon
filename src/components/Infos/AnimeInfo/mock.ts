interface mockTypes {
  format: string;
  duration: number;
  status: string;
  seasonYear: number;
  season: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  duration: 24,
  averageScore: 85,
  meanScore: 88,
  favourites: 500,
  format: "TV",
  genres: ["comedy", "horror"],
  popularity: 1000,
  season: "winter",
  seasonYear: 2024,
  source: "Light Novel",
  status: "Airing",
};
