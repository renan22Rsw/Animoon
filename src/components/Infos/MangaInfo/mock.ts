interface mockTypes {
  format: string;
  status: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  averageScore: 80,
  meanScore: 85,
  favourites: 1200,
  format: "Manga",
  genres: ["action", "adventure"],
  popularity: 1000,
  source: "Original",
  status: "Releasing",
};
