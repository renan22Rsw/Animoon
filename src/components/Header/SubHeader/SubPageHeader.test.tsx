import { render, screen } from "@testing-library/react";
import SubPageHeader from "./SubPageHeader";
import { mock } from "./mock";

describe("SubPageHeader component", () => {
  it("should render SubPageHeader images", () => {
    render(<SubPageHeader {...mock} />);
    const desktopImage = screen.getByTestId("desktop-version");
    const mobileImage = screen.getByTestId("mobile-version");
    expect(desktopImage).toBeInTheDocument();
    expect(mobileImage).toBeInTheDocument();
    expect(desktopImage).toHaveAttribute("width", "130");
    expect(mobileImage).toHaveAttribute("width", "130");
    expect(desktopImage).toHaveAttribute("height", "130");
    expect(mobileImage).toHaveAttribute("height", "130");
    expect(desktopImage).toHaveStyle({
      height: "345px",
      width: "230px",
    });
    expect(mobileImage).toHaveStyle({
      height: "345px",
      width: "230px",
    });
    expect(desktopImage).toHaveAttribute("fetchpriority", "high");
    expect(mobileImage).toHaveAttribute("fetchpriority", "high");
  });

  it("should render SubPageHeader - Desktop Version", () => {
    render(<SubPageHeader {...mock} />);
    const titles = screen.getAllByText(mock.name);
    expect(titles[0]).toBeInTheDocument();
    expect(titles[1]).toBeInTheDocument();

    const nativeNameDesktop = screen.getByTestId("nativeName-desktop");
    expect(nativeNameDesktop).toBeInTheDocument();
    expect(nativeNameDesktop).toHaveTextContent(mock.nativeName);

    const ageDesktop = screen.getByTestId("desktop-age");
    expect(ageDesktop).toBeInTheDocument();
    expect(ageDesktop).toHaveTextContent(mock.age);

    const genderDesktop = screen.getByTestId("desktop-gender");
    expect(genderDesktop).toBeInTheDocument();
    expect(genderDesktop).toHaveTextContent(mock.gender);

    const birthdayDesktop = screen.getByTestId("desktop-birthday");
    expect(birthdayDesktop).toBeInTheDocument();
    expect(birthdayDesktop).toHaveTextContent(`${mock.month}/${mock.day}`);

    const hometownDesktop = screen.getByTestId("desktop-hometown");
    expect(hometownDesktop).toBeInTheDocument();
    expect(hometownDesktop).toHaveTextContent(mock.hometown);

    const bloodTypeDesktop = screen.getByTestId("desktop-bloodType");
    expect(bloodTypeDesktop).toBeInTheDocument();
    expect(bloodTypeDesktop).toHaveTextContent(mock.bloodType);
  });

  it("should render SubPages - Mobile Version", () => {
    render(<SubPageHeader {...mock} />);
    const titles = screen.getAllByText(mock.name);
    expect(titles[0]).toBeInTheDocument();
    expect(titles[1]).toBeInTheDocument();

    const nativeNameMobile = screen.getByTestId("nativeName-mobile");
    expect(nativeNameMobile).toBeInTheDocument();
    expect(nativeNameMobile).toHaveTextContent(mock.nativeName);

    const ageMobilie = screen.getByTestId("mobile-age");
    expect(ageMobilie).toBeInTheDocument();
    expect(ageMobilie).toHaveTextContent(mock.age);

    const genderMobile = screen.getByTestId("mobile-gender");
    expect(genderMobile).toBeInTheDocument();
    expect(genderMobile).toHaveTextContent(mock.gender);

    const birthdayMobile = screen.getByTestId("mobile-birthday");
    expect(birthdayMobile).toBeInTheDocument();
    expect(birthdayMobile).toHaveTextContent(`${mock.month}/${mock.day}`);

    const hometownMobile = screen.getByTestId("mobile-hometown");
    expect(hometownMobile).toBeInTheDocument();
    expect(hometownMobile).toHaveTextContent(mock.hometown);

    const bloodTypeMobile = screen.getByTestId("mobile-bloodType");
    expect(bloodTypeMobile).toBeInTheDocument();
    expect(bloodTypeMobile).toHaveTextContent(mock.bloodType);
  });

  it("should handle missing optional props gracefully", () => {
    const incompleteMock = {
      ...mock,
      hometown: undefined,
      yearsActive: undefined,
    };
    render(<SubPageHeader {...incompleteMock} />);
    expect(screen.queryByTestId("desktop-hometown")).toBeNull();
    expect(screen.queryByText("Years active")).toBeNull();
  });

  it("should render the TextArea with the description prop", () => {
    render(<SubPageHeader {...mock} />);
    const textAreaDesktop = screen.getAllByTestId("text-area");
    expect(textAreaDesktop[0]).toBeInTheDocument();
    expect(textAreaDesktop[0]).toHaveTextContent(mock.description);

    expect(textAreaDesktop[1]).toBeInTheDocument();
    expect(textAreaDesktop[1]).toHaveTextContent(mock.description);
  });

  it("should match a snapshot", () => {
    const { container } = render(<SubPageHeader {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
