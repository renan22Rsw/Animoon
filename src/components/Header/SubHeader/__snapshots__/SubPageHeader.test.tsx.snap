// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubPageHeader component should match a snapshot 1`] = `
<div>
  <div>
    <div
      class="bg-[#1C1C1C] pb-[30px] pt-[20px] h-[200px] text-[#9FADB3]  "
    >
      <div
        class="max-w-[1200px]  hidden m-8 md:flex md:items-center lg:mx-10 2xl:mx-auto"
      >
        <img
          alt="Anime-Image"
          class="rounded-md "
          data-nimg="1"
          data-testid="desktop-version"
          decoding="async"
          fetchpriority="high"
          height="300"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=100"
          srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=100 2x"
          style="color: transparent; height: auto; width: 230px; object-fit: cover;"
          width="300"
        />
        <div
          class="my-4 mx-8  h-[300px]"
        >
          <h2
            class="text-[#ADC0D2] text-3xl font-bold py-2"
          >
            Naruto
          </h2>
          <p
            class="md:text-sm lg:text-sm"
            data-testid="nativeName-desktop"
          >
            Naruto Uzumaki, 
            Nine-Tails Jinchuuriki,  Number One Unpredictable
          </p>
          <div
            class="relative top-[100px]"
          >
            <h4
              data-testid="desktop-birthday"
            >
              Birthday: 
              10/10
            </h4>
            <h4
              data-testid="desktop-age"
            >
              Age: 
              17
            </h4>
            <h4
              data-testid="desktop-gender"
            >
              Gender: 
              Male
               
            </h4>
            <h4>
              Years active: 
              1996
            </h4>
            <h4
              data-testid="desktop-hometown"
            >
              Hometown: 
              Saitama Prefecture, Japan
               
            </h4>
            <h4
              data-testid="desktop-bloodType"
            >
              Blood Type: 
              B
            </h4>
            <div
              style="display: flex;"
            >
              <textarea
                aria-label="empty textarea"
                class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
                cols="100"
                data-testid="text-area"
                disabled=""
                placeholder="Empty"
                rows="1"
                style="height: 0px; overflow: hidden;"
              >
                Born in Konohagakure, a ninja village hidden in the leaves
              </textarea>
              <textarea
                aria-hidden="true"
                class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center px-4 md:hidden  "
      >
        <h2
          class="text-[#ADC0D2] text-2xl font-bold py-2"
        >
          Naruto
        </h2>
        <p
          data-testid="nativeName-mobile"
        >
          Naruto Uzumaki, 
          Nine-Tails Jinchuuriki,  Number One Unpredictable
        </p>
        <div
          class="flex justify-center p-4  my-8"
        >
          <img
            alt="Anime-Image"
            class="rounded-md "
            data-nimg="1"
            data-testid="mobile-version"
            decoding="async"
            fetchpriority="high"
            height="130"
            src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75"
            srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=256&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=384&q=75 2x"
            style="color: transparent; height: auto; width: 230px; object-fit: cover;"
            width="130"
          />
        </div>
        <div
          class="text-start mx-4"
        >
          <h4
            data-testid="mobile-birthday"
          >
            Birthday: 
            10/10
          </h4>
          <h4
            data-testid="mobile-age"
          >
            Age: 
            17
          </h4>
          <h4
            data-testid="mobile-gender"
          >
            Gender: 
            Male
             
          </h4>
          <h4>
            Years active: 
            1996
          </h4>
          <h4
            data-testid="mobile-hometown"
          >
            Hometown: 
            Saitama Prefecture, Japan
             
          </h4>
          <h4
            data-testid="mobile-bloodType"
          >
            Blood Type: 
            B
          </h4>
          <div
            style="display: flex;"
          >
            <textarea
              aria-label="empty textarea"
              class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
              cols="100"
              data-testid="text-area"
              disabled=""
              placeholder="Empty"
              rows="1"
              style="height: 0px; overflow: hidden;"
            >
              Born in Konohagakure, a ninja village hidden in the leaves
            </textarea>
            <textarea
              aria-hidden="true"
              class=" w-full text-sm font-normal bg-transparent  leading-normal  resize-none"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
